src/Common/src/System/Data/Common/AdapterUtil.cs

```diff
@@ -361,24 +361,33 @@ internal static Exception StreamClosed([CallerMemberName] string method = "")
 
         internal static string BuildQuotedString(string quotePrefix, string quoteSuffix, string unQuotedString)
         {
-            var resultString = new StringBuilder();
+            var resultString = new StringBuilder(unQuotedString.Length + quoteSuffix.Length + quoteSuffix.Length);
+            AppendQuotedString(resultString, quotePrefix, quoteSuffix, unQuotedString);
```

Comment:
Isn't the result of AppendQuotedString same value as resultString.ToString()?

Reply:
The output is the same, The new path doesn't allocate a second stringbuilder or a string for that segment it just writes everything to the stringbuilder, That also allows the replace call to avoid a string allocation because it can be done in the internal buffer.

Reply:
My idea was  that BuildQuotedString method can look like this:
`internal static string BuildQuotedString(string quotePrefix, string quoteSuffix, string unQuotedString)
        {
            var resultString = new StringBuilder(unQuotedString.Length + quoteSuffix.Length + quoteSuffix.Length);
            
            return AppendQuotedString(resultString, quotePrefix, quoteSuffix, unQuotedString);
        }`

But maybe I don't see something.

Reply:
It could but that would change AppendQuotedString to be allocating when there's no need to. If you want to produce a multipart name you could call AppendQuotedString up to 4 times with . separators in between them. I chose to provide flexibility by having one method responsible for the lifetime of the builder and the other simply using it.